constraint/pkg/instrumentation/instrumentation.go

```diff
@@ -0,0 +1,24 @@
+// package instrumentation defines primitives to
+// support more observability throughout the contraint framework.
+package instrumentation
+
+// NamedStats covers primitive for many heterogenoues statstics and metrics
+// to be represented to consumers.
+type NamedStats struct {
+	StatsName string
+	Stats     []*Stat
+}
+
+// Stat is a Name, Value, Description tuple that may contain
+// metrics or aggregations of metrics.
+type Stat struct {
+	Name        string
+	Value       string
+	Description string
```

Comment:
Description could be valuable for giving drivers a way to document their responses